.pre-commit-config-cpp.yaml

```diff
@@ -0,0 +1,26 @@
+# This file contains pre-commit hooks for checking the C++ code. Since these
+# hooks require tools that may not be trivial to install (unlike the Python
```

Comment:
`pre-commit`'s web page suggests that it handles the installation of even non-trivial tools. Does it not support `clang-format` in that way. It would be nice to have the whole config together.

Reply:
I looked at the [list of pre-commit hooks](https://pre-commit.com/hooks.html) that wrap clang-format. I think all of them require clang-format to be installed on the system. There is one version that can use a Docker container, but `docker` isn't any easier to fulfill as a requirement than `clang`. https://gitlab.com/daverona/pre-commit/cpp